### ISSUE TITLE
Consistency with BfasstException

### ISSUE THREAD

**Issue by jgoeders** [byuccl/bfasst#111]
It looks like BfasstException was added to the code base, but I'm not sure we are using it consistently.  In some cases it looks like we are raising exceptions, and in others we are returning error statuses:

```
        # Run Icecube 2 LSE synthesis
        try:
            status = self.run_sythesis(prj_path, log_path)
        except BfasstException as e:
            # If generic error, see if log has something more specific
            if e.error == OptStatus.ERROR:
                new_status = self.check_opt_log(log_path)
                return new_status
            raise e
```

**Comment by reillymck** [comment]
OK, so basically, I am the one that implemented the exceptions. I did a quick patch where I changed the status base class to raise an exception if an error status was created. I did this because I didn't want to change every line of status code (it's a lot). Some places needed to catch the exception like demonstrated above, so I changed those to try catch. Other wise it's brought to the top level. With the multithreading code I added a catch clause so the whole run wouldn't fail.
